components/core/src/clp/ffi/ir_stream/ir_unit_deserialization_methods.cpp

```diff
@@ -460,6 +461,24 @@ auto deserialize_value_and_construct_node_id_value_pairs(
 }
 }  // namespace
 
+auto get_ir_unit_type_from_tag(encoded_tag_t tag) -> IrUnitType {
+    // First, we check the tags that have one-to-one IR unit mapping
+    if (cProtocol::Eof == tag) {
+        return IrUnitType::EndOfStream;
+    }
+    if (cProtocol::Payload::UtcOffsetChange == tag) {
+        return IrUnitType::UtcOffsetChange;
+    }
+
+    // Then, check tags that may match any byte within a continuous range
+    if ((tag & cProtocol::Payload::SchemaTreeNodeMask) == cProtocol::Payload::SchemaTreeNodeMask) {
+        return IrUnitType::SchemaTreeNodeInsertion;
+    }
+
+    // If not match, it is assumed to be a log event
+    return IrUnitType::LogEvent;
+}
```

Comment:
Shouldn't send a warning. For any tag that wasn't categorized, we assume it's a tag for log event. Since a log event IR unit may start with too many possible tags, adding an addition check here can be very expensive. Unsupported tags will be caught using the deserialization method anyway.